### PR TITLE
OCPBUGS-20347: UPSTREAM: <carry>: SCC RBAC

### PR DESCRIPTION
**Pull request by awgreene**
After reducing the RBAC granted to the OLM ServiceAccount in an earlier commit, this commit introduces RBAC so pods that use the OLM ServiceAccount will qualify to use the restricted-v2 SCC.

### CODE REVIEW

**Comment by awgreene** [comment]
@ncdc I'm confused, this is a downstream only commit, could you clarify when to use `carry` vs `drop`?

**Comment by ncdc** [comment]
Sure! The terminology is a bit "weird":
* carry: we need this downstream and it's not upstream
* drop: use for `go mod vendor` commits, anything having to do with generating files (because we'll redo these each rebase), or anything we don't intend to carry forward in the next rebase

**Comment by awgreene** [comment]
Thanks for the definition @ncdc! I think I've got the commit/PR in the correct state based on your guidance.

**Comment by grokspawn** [comment]
> Sure! The terminology is a bit "weird":

It helped me to think of the "me" perspective as upstream.  Then these are all messages to upstream about what to do with what's downstream and the semantics make sense.

**Comment by jianzhangbjz** [comment]
Test fail, details: https://issues.redhat.com/browse/OCPBUGS-20347
/hold

File: manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml [hunk 1]
```diff
@@ -22,6 +22,14 @@ rules:
     verbs: ["*"]
   - nonResourceURLs: ["*"]
     verbs: ["*"]
+  - apiGroups:
```
Comment (jianzhangbjz) [inline manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml:25]: Should also remove this:
```console
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
 ```

File: manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml [hunk 2]
```diff
@@ -22,6 +22,14 @@ rules:
     verbs: ["*"]
```
Comment (jianzhangbjz) [inline manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml:22]: This still exists, it still has high access permission and can get all SCC.
```console
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
 ```

Comment (awgreene) [reply manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml:22]: @jianzhangbjz we need [this PR](https://github.com/openshift/operator-framework-olm/pull/600) which contains [this commit](https://github.com/openshift/operator-framework-olm/pull/600/commits/43f13e1c239bc980642259ca29b624c15a35db23).

Comment (awgreene) [reply manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml]: I've pulled that PR into this commit.

**Comment by awgreene** [comment]
@stevekuznetsov is going to pull this commit into https://github.com/openshift/operator-framework-olm/pull/600, closing this.

File: manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml (hunk 2 above)
Comment (jianzhangbjz) [reply manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml]: Got it, thanks!

**Pull request closed without merging**